pallets/ocex/src/tests.rs

```diff
@@ -20,113 +20,605 @@ use frame_support::{
+    		fees: bounded_vec![],
+
+		};
+		assert_noop!(
+			OCEX::submit_snapshot(
+				Origin::signed(account_id.into()),
+				snapshot,
+				sig.clone().into()
+			), 
+			Error::<Test>::SenderIsNotAttestedEnclave
+		);
+	});
+}
+
+#[test]
+fn test_submit_snapshot_snapshot_nonce_error(){
+	let account_id = create_account_id();
+	let payl: [u8; 64] = [0; 64];
+	let sig = sp_core::sr25519::Signature::from_raw(payl);
+	new_test_ext().execute_with(||{
+		let mmr_root: H256 = create_mmr_with_one_account();
+		let mut snapshot = EnclaveSnapshot::<AccountId32, Balance, WithdrawalLimit, AssetsLimit>{
+			snapshot_number: 1,
+    		merkle_root: mmr_root,
+			withdrawals: bounded_vec![],
+    		fees: bounded_vec![],
+
+		};
+		assert_ok!(
+			OCEX::insert_enclave(
+				Origin::root(),
+				account_id.clone().into()
+			)
+		);
+		assert_noop!(
+			OCEX::submit_snapshot(
+				Origin::signed(account_id.into()),
+				snapshot,
+				sig.clone().into()
+			), 
+			Error::<Test>::SnapshotNonceError
+		);
+	});
+}
+
+#[test]
+fn test_submit_snapshot_enclave_signature_verification_failed(){
+	let account_id = create_account_id();
+	let payl: [u8; 64] = [0; 64];
+	let sig = sp_core::sr25519::Signature::from_raw(payl);
+	new_test_ext().execute_with(||{
+		let mmr_root: H256 = create_mmr_with_one_account();
+		let mut snapshot = EnclaveSnapshot::<AccountId32, Balance, WithdrawalLimit, AssetsLimit>{
+			snapshot_number: 0,
+    		merkle_root: mmr_root,
+			withdrawals: bounded_vec![],
+    		fees: bounded_vec![],
+
+		};
+		assert_ok!(
+			OCEX::insert_enclave(
+				Origin::root(),
+				account_id.clone().into()
+			)
+		);
+		assert_noop!(
+			OCEX::submit_snapshot(
+				Origin::signed(account_id.into()),
+				snapshot,
+				sig.clone().into()
+			), 
+			Error::<Test>::EnclaveSignatureVerificationFailed
+		);
+	});
+}
+
+#[test]
+fn test_submit_snapshot(){
+	let account_id = create_account_id(); 
+	const PHRASE: &str =
+		"news slush supreme milk chapter athlete soap sausage put clutch what kitten";
+	let public_key_store = KeyStore::new();
+	let public_key = SyncCryptoStore::sr25519_generate_new(
+		&public_key_store,
+		KEY_TYPE,
+		Some(&format!("{}/hunter1", PHRASE)),
+	)
+	.expect("Unable to create sr25519 key pair");
+	let mut t = new_test_ext();
+	t.register_extension(KeystoreExt(Arc::new(public_key_store)));
+	t.execute_with(||{
+		let mmr_root: H256 = create_mmr_with_one_account();
+		let mut snapshot = EnclaveSnapshot::<AccountId32, Balance, WithdrawalLimit, AssetsLimit>{
+			snapshot_number: 0,
+    		merkle_root: mmr_root,
+			withdrawals: bounded_vec![],
+    		fees: bounded_vec![],
+
+		};
+		assert_ok!(
+			OCEX::insert_enclave(
+				Origin::root(),
+				account_id.clone().into()
+			)
+		);
+		let bytes = snapshot.encode();
+		let signature = public_key.sign(KEY_TYPE, &bytes).unwrap();
+		
+		assert_ok!(
+			OCEX::submit_snapshot(
+				Origin::signed(account_id.into()),
+				snapshot,
+				signature.clone().into()
+			),
+		);
+	})
+}
```

Comment:
All data changes should be asserted [here](https://github.com/Polkadex-Substrate/Polkadex/blob/d74dfd99a3dad44b2ae60a29d84bf0e7fdeff2cb/pallets/ocex/src/lib.rs#L385-L390).

Reply:
Updated ✅

Reply:
I can see that asserted the present of keys in collection but real data change is not asserted.

Reply:
Hey @serhii-temchenko could you elaborate this?

Reply:
Updated ✅